tests/5.1/teams/test_target_set_num_teams.c

```diff
@@ -0,0 +1,51 @@
+//===--------------------- test_target_set_num_teams.c ----------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test uses the omp_set_num_teams utility to set the default number of
+// teams allocated for team clauses. The test runs teams and checks to see if
+// the number allocated is correct and reflective of the new default set. It
+// then checks to see if an explicitly stated num_teams clause overrides
+// the omp_set_num_teams properly.
+//
+//===------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+int main() {
+
+	OMPVV_TEST_OFFLOADING;
+
+	int errors = 0; 
+	int num_teams = 0;
+
+        #pragma omp target
+        {
+	  omp_set_num_teams(4);
+        }
+	
+	#pragma omp target teams map(tofrom: num_teams)
+	{
+		if (omp_get_team_num() == 0 && omp_get_thread_num() == 0) {
+			num_teams = omp_get_num_teams();
```

Comment:
(**GCC** supports **omp_set_num_teams** since GCC 12, but:)

This **fails with GCC** (tried: mainline/GCC 14) with:
```C
test_target_set_num_teams.c: In function ‘main’:
test_target_set_num_teams.c:32:48: error: OpenMP runtime API call ‘omp_get_thread_num’ strictly nested in a ‘teams’ region
   32 |                 if (omp_get_team_num() == 0 && omp_get_thread_num() == 0) {
      |                                                ^~~~~~~~~~~~~~~~~~~~
test_target_set_num_teams.c:42:48: error: OpenMP runtime API call ‘omp_get_thread_num’ strictly nested in a ‘teams’ region
   42 |                 if (omp_get_team_num() == 0 && omp_get_thread_num() == 0) {
      |                                                ^~~~~~~~~~~~~~~~~~~~
```

Reply:
@tob2 , thanks for testing it. This is an easy fix.